src/display.c

```diff
@@ -80,18 +80,30 @@ int display_configure(void)
 
     // Detect if display is connected
     int timing_cnt, color_cnt;
-    int hpd = ret = dcp_ib_get_hpd(iboot, &timing_cnt, &color_cnt);
-
-    if (hpd < 0) {
+    int hpd = 0, retries = 0;
+
+    /* After boot DCP does not immediately report a connected display. Retry getting display
+     * information for 2 seconds.
+     */
+    while (retries++ < 20) {
+        hpd = dcp_ib_get_hpd(iboot, &timing_cnt, &color_cnt);
+        if (hpd < 0) {
+            printf("display: try:%d failed to get display status\n", retries);
+            ret = hpd;
+        } else if (hpd && timing_cnt && color_cnt) {
+            ret = 0;
+            printf("display: try:%d connected:%d timing_cnt:%d color_cnt:%d\n", retries, hpd,
+                   timing_cnt, color_cnt);
+            break;
+        }
+        if (retries < 20)
+            mdelay(100);
+    }
+    if (ret < 0) {
```

Comment:
Shouldn't this also bail if hpd == 0? That's the display not connected case. Right now, as far as I can tell that will fall out of the loop and be treated as success.

Reply:
you're right, I'll push a fix